public/javascripts/janus.js

```diff
@@ -279,4 +279,33 @@ jQuery(function($){
         }
     });
 
+    // auto-logout switch (persisted with local storage)
+    const LOCAL_STORAGE_KEY__AUTO_LOGOUT = "autoLogout"
+    let shouldAutoLogout = localStorage.getItem(LOCAL_STORAGE_KEY__AUTO_LOGOUT) === "true";
+    const autoLogoutSwitch = $("#auto_logout_switch")
+    autoLogoutSwitch.prop('checked', shouldAutoLogout)
+    autoLogoutSwitch.on("change", () => {
+        shouldAutoLogout = event.target.checked;
+        localStorage.setItem(LOCAL_STORAGE_KEY__AUTO_LOGOUT, shouldAutoLogout.toString());
+    });
+
+    // auto-logout for console buttons
+    $("a[href*='/console?permissionId=']").each(function(_, el){
+        const targetHref = el.href;
+        el.removeAttribute("href");
```

Comment:
I thought removing the `href` attribute disabled the link, which would mean it isn't focusable? If so, is it better to use preventDefault in the handler and leave the href attribute present? This also means things like dragging the link to another tab instead of clicking it would still work (with the caveat that this auto-logout would not work that way*).

* I think these kind of UX-problem edge cases are a necessary evil of this approach though!

Reply:
excellent point, thanks @adamnfish - will do that now